---
title: '12\.Phylogenetic Diversity - Communities'
author: "Dustin Brewer; Z620: Quantitative Biodiversity, Indiana University"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
geometry: margin=2.54cm
---

## OVERVIEW
Complementing taxonomic measures of $\alpha$- and $\beta$-diversity with evolutionary information yields insight into a broad range of biodiversity issues including conservation, biogeography, and community assembly. 
In this worksheet, you will be introduced to some commonly used methods in phylogenetic community ecology.

After completing this assignment you will know how to:

1.  incorporate an evolutionary perspective into your understanding of community ecology
2.  quantify and interpret phylogenetic $\alpha$- and $\beta$-diversity
3.  evaluate the contribution of phylogeny to spatial patterns of biodiversity

## Directions:
1. In the Markdown version of this document in your cloned repo, change "Student Name" on line 3 (above) with your name.
2. Complete as much of the worksheet as possible during class.
3. Use the handout as a guide; it contains a more complete description of data sets along with examples of proper scripting needed to carry out the exercises.
4. Answer questions in the  worksheet.
Space for your answers is provided in this document and is indicated by the ">" character.
If you need a second paragraph be sure to start the first line with ">".
You should notice that the answer is highlighted in green by RStudio (color may vary if you changed the editor theme). 
5. Before you leave the classroom today, it is *imperative* that you **push** this file to your GitHub repo, at whatever stage you are. This will enable you to pull your work onto your own computer.
6. When you have completed the worksheet, **Knit** the text and code into a single PDF file by pressing the `Knit` button in the RStudio scripting panel.
This will save the PDF output in your '8.BetaDiversity' folder.
7. After Knitting, please submit the worksheet by making a **push** to your GitHub repo and then create a **pull request** via GitHub.
Your pull request should include this file *12.PhyloCom_Worksheet.Rmd* and the PDF output of `Knitr` (*12.PhyloCom_Worksheet.pdf*).

## 1) SETUP

Typically, the first thing you will do in either an R script or an RMarkdown file is setup your environment. 
This includes things such as setting the working directory and loading any packages that you will need.

In the R code chunk below, provide the code to:  
1. clear your R environment,  
2. print your current working directory,  
3. set your working directory to your `/Week7-PhyloCom` folder,  
4. load all of the required R packages (be sure to install if needed), and  
5. load the required R source file.

```{r}
rm(list = ls())
getwd()



 library("picante", lib.loc="~/R/win-library/3.5")
library("ape", lib.loc="~/R/win-library/3.5")
 library("seqinr", lib.loc="~/R/win-library/3.5")
library("vegan", lib.loc="~/R/win-library/3.5")
library("fossil", lib.loc="~/R/win-library/3.5")
library("reshape", lib.loc="~/R/win-library/3.5")
library("simba", lib.loc="~/R/win-library/3.5")



source("./bin/MothurTools.R")

```



## 2) DESCRIPTION OF DATA

### need to discuss data set from spatial ecology!
 
In 2013 we sampled > 50 forested ponds in Brown County State Park, Yellowwood State Park, and Hoosier National Forest in southern Indiana. 
In addition to measuring a suite of geographic and environmental variables, we characterized the diversity of bacteria in the ponds using molecular-based approaches.
Specifically, we amplified the 16S rRNA gene (i.e., the DNA sequence) and 16S rRNA transcripts (i.e., the RNA transcript of the gene) of bacteria.
We used a program called `mothur` to quality-trim our data set and assign sequences to operational taxonomic units (OTUs), which resulted in a site-by-OTU matrix.  
In this module we will focus on taxa that were present (i.e., DNA), but there will be a few steps where we need to parse out the transcript (i.e., RNA) samples.
See the handout for a further description of this week's dataset.

## 3) LOAD THE DATA

In the R code chunk below, do the following:  
1. load the environmental data for the Brown County ponds (*20130801_PondDataMod.csv*),  
2. load the site-by-species matrix using the `read.otu()` function,  
3. subset the data to include only DNA-based identifications of bacteria,  
4. rename the sites by removing extra characters,  
5. remove unnecessary OTUs in the site-by-species, and   
6. load the taxonomic data using the `read.tax()` function from the source-code file.

```{r}
env <- read.table("data/20130801_PondDataMod.csv", sep = ",", header = TRUE)
env <- na.omit(env)

str(env)

#Load site by species matrix
comm <- read.otu(shared = "./data/INPonds.final.rdp.shared", cutoff = "1")

#Select DNA, disregard cDNA
comm <- comm[grep("*-DNA", rownames(comm)),]

#Perform replacement of all matches
rownames(comm) <- gsub("\\-DNA", "", rownames(comm))
rownames(comm) <- gsub("\\_", "", rownames(comm))

#remove sites not in the environmental data set
comm <- comm[rownames(comm) %in% env$Sample_ID,]

#remove zero-abundance OTUs from data set
comm <- comm[ , colSums(comm) > 0]

#From the source code, take a look at taxonomic information associated with OTU data
tax <- read.tax(taxonomy = "./data/INPonds.final.rdp.1.cons.taxonomy")

```

Next, in the R code chunk below, do the following:  
1. load the FASTA alignment for the bacterial operational taxonomic units (OTUs),  
2. rename the OTUs by removing everything before the tab (\\t) and after the bar (|),  
3. import the *Methanosarcina* outgroup FASTA file,  
4. convert both FASTA files into the DNAbin format and combine using `rbind()`,  
5. visualize the sequence alignment,  
6. using the alignment (with outgroup), pick a DNA substitution model, and create a phylogenetic distance matrix,  
7. using the distance matrix above, make a neighbor joining tree,  
8. remove any tips (OTUs) that are not in the community data set,  
9. plot the rooted tree.

```{r}
#import the alignment file
ponds.cons <- read.alignment(file = "./data/INPonds.final.rdp.1.rep.fasta", format = "fasta")

#rename OTUs
ponds.cons$nam <- gsub("\\|.*$", "", gsub("^.*?\t", "", ponds.cons$nam))

#import the outgroup (archaea)
outgroup <- read.alignment(file = "./data/methanosarcina.fasta", format = "fasta")

#Convert outgroup alignment file to DNAbin
DNAbin <- rbind(as.DNAbin(outgroup), as.DNAbin(ponds.cons))

#have a look at the alignment
image.DNAbin(DNAbin, show.labels=T, cex.lab = 0.05, las = 1)

#make a distance matrix, using 'jukes cantor' model
seq.dist.jc <- dist.dna(DNAbin, model = "JC", pairwise.deletion = FALSE)

#Use the jukes cantor object and make a neighbor-joining tree
phy.all <- bionj(seq.dist.jc)

#Drop tips associated with zero abundance taxa
phy <- drop.tip(phy.all, phy.all$tip.label[!phy.all$tip.label %in% 
              c(colnames(comm), "Methanosarcina")])

#ID outgroup sequence
outgroup <- match("Methanosarcina", phy$tip.label)

#Root the tree
phy <- root(phy, outgroup, resolve.root = TRUE)

#Plot the rooted tree
par(mar = c(1, 1, 2, 1) + 0.1)
plot.phylo(phy, main = "Neighbor Joining Tree", "phylogram", show.tip.label = FALSE, 
           use.edge.length = FALSE, direction = "right", cex = 0.6, label.offset = 1)



```

## 4) PHYLOGENETIC ALPHA DIVERSITY

### A.  Faith's Phylogenetic Diversity (PD)
In the R code chunk below, do the following:  
1. calculate Faith's D using the `pd()` function.

```{r}
#Calculate Faith's D and species richness
pd <- pd(comm, phy, include.root = FALSE)
pd


```

In the R code chunk below, do the following:  
1. plot species richness (S) versus phylogenetic diversity (PD),  
2. add the trend line, and  
3. calculate the scaling exponent. 

```{r}
#Make a biplot of species richness and phylogenetic diversity, using a natural logarithm on both axes so that they become comparable (so slope of relationship gives 'power law relationship) 
par(mar = c(5, 5, 4, 1) + 0.1)
plot(log(pd$S), log(pd$PD), 
     pch = 20, col = "red", las = 1,
     xlab = "ln(S)", ylab = "ln(PD)", cex.main = 1,
     main="Phylodiversity (PD) vs. Taxonomic richness (S)")

# Test the power-law relationship
fit <- lm('log(pd$PD) ~ log(pd$S)')
abline(fit, col = "red", lw = 2)
exponent <- round(coefficients(fit)[2], 2)
legend("topleft", legend=paste("Scaling exponent = ", exponent, sep = ""), 
       bty = "n", lw = 2, col = "red") 

```

***Question 1***: Answer the following questions about the PD-S pattern.    
a.  Based on how PD is calculated, why should this metric be related to taxonmic richness? 
b.  Describe the relationship between taxonomic richness and phylodiversity.
c.  When would you expect these two estimates of diversity to deviate from one another? 
d.  Interpret the significance of the scaling PD-S scaling exponent.

> ***Answer 1a***:  PD does not weight the branches of common species more than the branches of rare species, just like a measure of species richness, so it makes sense that the measures will be related. As there are more species at a site, there will be more branch length measured by PD. 
> ***Answer 1b***:  There is positive, non-linear relationship between PD and S. As species richness increases, so does phylogenetic diversity (but PD does not increase as fast). 
> ***Answer 1c***: If there were many closely related species in an area, it seems that species richness and phylogenetic diversity there would diverge.   
> ***Answer 1d***: Because the scaling exponent is less then 1, I take that to mean that Phylogenetic Diversity increases at a slower rate than species richness.

**i. Randomizations and Null Models**  

In the R code chunk below, do the following:  
1. estimate the standardized effect size of PD using the `richness` randomization method.  

```{r}
#randomize phylogenetic diversity to create a null model that our data can be compared to, using the picante null model 'richness.'

ses.pd <- ses.pd(comm[1:2,], phy, null.model = "richness", runs = 25, 
                 include.root = FALSE)

#do same, but with null model of 'taxa labels'
ses.pd2 <- ses.pd(comm[1:2,], phy, null.model = "taxa.labels", runs = 25, include.root = FALSE)
ses.pd2

#now with 'frequency'
ses.pd3 <- ses.pd(comm[1:2,], phy, null.model = "frequency", runs = 25, include.root = FALSE)
ses.pd3

?ses.pd()

```

***Question 2***: Using `help()` and the table above, run the `ses.pd()` function using two other null models and answer the following questions:

a.  What are the null and alternative hypotheses you are testing via randomization when calculating `ses.pd`?
b.  How did your choice of null model influence your observed ses.pd values?  Explain why this choice affected or did not affect the output. 

> ***Answer 2a***:  The null hypothesis being tested when using ses.pd is that there is no difference between the randomized communities generated by the null model and the one that you sampled/are interested in. The alternative hypothesis is that there is a difference.  
> ***Answer 2b***:  The 'richness' null model was not significantly different than BC001 or BC002, nor was the 'taxa.labels' null model. The 'frequency' null model was not significantly different than BC001, but it was signficantly different than BC002. I think that BC002 was significantly different than the 'frequency' null model because in that model, sample species richness is not maintained and BC002 has an above average species richness. 

### B.  Phylogenetic Dispersion Within a Sample
Another way to assess phylogenetic $\alpha$-diversity is to look at dispersion within a sample. 

**i. Phylogenetic Resemblance Matrix**

In the R code chunk below, do the following:  
1. calculate the phylogenetic resemblance matrix for taxa in the Indiana ponds data set.

```{r}
#Create a phylogenetic distance matrix (which is based on distance between tax on the tree)
phydist <- cophenetic.phylo(phy)

```

**ii.  Net Relatedness Index (NRI)**

In the R code chunk below, do the following:  
1. Calculate the NRI for each site in the Indiana ponds data set. 

```{r}
#Estimate standardized effect size of Net Relatedness Index via randomizations
ses.mpd <- ses.mpd(comm, phydist, null.model = "taxa.labels", 
                   abundance.weighted = FALSE, runs = 25)

NRI <- as.matrix(-1 * ((ses.mpd[,2] - ses.mpd[,3]) / ses.mpd[,4]))
rownames(NRI) <- row.names(ses.mpd)
colnames(NRI) <- "NRI"

NRI


#Do the same, but with abundance.weighted = TRUE
ses.mpd2 <- ses.mpd(comm, phydist, null.model = "taxa.labels", 
                   abundance.weighted = TRUE, runs = 25)

NRI2 <- as.matrix(-1 * ((ses.mpd2[,2] - ses.mpd2[,3]) / ses.mpd2[,4]))
rownames(NRI) <- row.names(ses.mpd)
colnames(NRI) <- "NRI2"

NRI2



```

**iii.  Nearest Taxon Index (NTI)**

In the R code chunk below, do the following:
1. Calculate the NTI for each site in the Indiana ponds data set. 

```{r}
# Estimate Standardized Effect Size of NRI via Randomization
ses.mntd <- ses.mntd(comm, phydist, null.model = "taxa.labels", 
                     abundance.weighted = FALSE, runs = 25)
# Calculate NTI
NTI <- as.matrix(-1 * ((ses.mntd[,2] - ses.mntd[,3]) / ses.mntd[,4]))
rownames(NTI) <- row.names(ses.mntd)
colnames(NTI) <- "NTI"

NTI


#Do same, but with abundance.weighted as TRUE
ses.mntd2 <- ses.mntd(comm, phydist, null.model = "taxa.labels", 
                     abundance.weighted = TRUE, runs = 25)
# Calculate NTI
NTI2 <- as.matrix(-1 * ((ses.mntd2[,2] - ses.mntd2[,3]) / ses.mntd2[,4]))
rownames(NTI) <- row.names(ses.mntd2)
colnames(NTI) <- "NTI2"

NTI2



```

***Question 3***: 

a. In your own words describe what you are doing when you calculate the NRI. 
b. In your own words describe what you are doing when you calculate the NTI. 
c. Interpret the NRI and NTI values you observed for this dataset. 
d. In the NRI and NTI examples above, the arguments "abundance.weighted = FALSE" means that the indices were calculated using presence-absence data.
Modify and rerun the code so that NRI and NTI are calculated using abundance data. 
How does this affect the interpretation of NRI and NTI?

> ***Answer 3a***:  When calculating NRI, I am trying to determine to what extent my sample is clustered or overdispersed based on a comparison with a null model. This comparison is made based upon pairwise comparisons between all taxa in a sample. 
> ***Answer 3b***:  The NTI is similar to the NRI, but is based on nearest neighbor distance, rather than by making pairwise comparisons with all taxa in the tree. Thus, this measure focuses more on clustering that occurs near the tips. 
> ***Answer 3c***:  Both the NTI and NRI values suggest that the bacterial communities tend to be overdispersed. 
> ***Answer 3d***:  Calculating NRI and NTI using abundance data rather than incidence data changes the interpretation. Rather than tending to indicate overdispersion, both NRI and NTI tend to indicate clustering for OTUs at sites. 

## 5) PHYLOGENETIC BETA DIVERSITY 

### A. Phylogenetically Based Community Resemblance Matrix
In the R code chunk below, do the following:  
1. calculate the phylogenetically based community resemblance matrix using Mean Pair Distance, and  
2. calculate the phylogenetically based community resemblance matrix using UniFrac distance.

```{r}
#mean pair distance
dist.mp <- comdist(comm, phydist)

#UniFrac distance
dist.uf <- unifrac(comm, phy)

```

In the R code chunk below, do the following:  
1. plot Mean Pair Distance versus UniFrac distance and compare.

```{r}
par(mar = c(5, 5, 2, 1) + 0.1)
plot(dist.mp, dist.uf, 
     pch = 20, col = "red", las = 1, asp = 1, xlim = c(0.15, 0.5), ylim = c(0.15, 0.5),
     xlab = "Mean Pair Distance", ylab = "UniFrac Distance") 
abline(b = 1, a = 0, lty = 2)
text(0.5, 0.47, "1:1")

```

***Question 4***: 

a. In your own words describe Mean Pair Distance, UniFrac distance, and the difference between them.
b. Using the plot above, describe the relationship between Mean Pair Distance and UniFrac distance. 
Note: we are calculating unweighted phylogenetic distances (similar to incidence based measures).
That means that we are not taking into account the abundance of each taxon in each site.
c. Why might MPD show less variation than UniFrac?

> ***Answer 4a***: Mean Pair Distance compares the average distance between taxa from two different samples. Unifrac Distance is different than Mean Pair Distance because it compares two samples by computing the proportion of the two trees that are shared with respect to branch topology. 
> ***Answer 4b***: There is much more variability in Unifrac distance (~0.2 to 0.45) than in Mean Pair Distance (~0.29 to 0.31). 
> ***Answer 4c***: MPD might show less variation than Unifrac because even if both samples have similar mean distances between pairs of taxa, the phylogenetic trees might vary more with respect to branch lengths, thus causing UniFrac to be more variable. 


### B. Visualizing Phylogenetic Beta-Diversity
Now that we have our phylogenetically based community resemblance matrix, we can visualize phylogenetic diversity among samples using the same techniques that we used in the $\beta$-diversity module from earlier in the course.

In the R code chunk below, do the following:  
1. perform a PCoA based on the UniFrac distances, and  
2. calculate the explained variation for the first three PCoA axes.

```{r}
#PCoA based on Unifrac distances
pond.pcoa <- cmdscale(dist.uf, eig = T, k = 3)
pond.pcoa
?cmdscale()

#Calculate explpained variation
explainvar1 <- round(pond.pcoa$eig[1] / sum(pond.pcoa$eig), 3) * 100
explainvar2 <- round(pond.pcoa$eig[2] / sum(pond.pcoa$eig), 3) * 100
explainvar3 <- round(pond.pcoa$eig[3] / sum(pond.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

```

Now that we have calculated our PCoA, we can plot the results.

In the R code chunk below, do the following:  
1. plot the PCoA results using either the R base package or the `ggplot` package,  
2. include the appropriate axes,   
3. add and label the points, and    
4. customize the plot. 

```{r}
# Define Plot Parameters
par(mar = c(5, 5, 1, 2) + 0.1)

# Initiate Plot
plot(pond.pcoa$points[ ,1], pond.pcoa$points[ ,2],
     xlim = c(-0.2, 0.2), ylim = c(-.16, 0.16),
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(pond.pcoa$points[ ,1], pond.pcoa$points[ ,2],
       pch = 19, cex = 3, bg = "gray", col = "gray")
text(pond.pcoa$points[ ,1], pond.pcoa$points[ ,2], 
     labels = row.names(pond.pcoa$points))

```

In the following R code chunk:
1. perform another PCoA on taxonomic data using an appropriate measure of dissimilarity, and
2. calculate the explained variation on the first three PCoA axes. 

```{r}
bc <- vegdist(comm, method = "bray", binary = FALSE, diag = FALSE, upper = FALSE, na.rm = TRUE)

#PCoA based on bray distances
tax.pcoa <- cmdscale(bc, eig = T, k = 3)
tax.pcoa


#Calculate explpained variation
xplainvar1 <- round(tax.pcoa$eig[1] / sum(tax.pcoa$eig), 3) * 100
xplainvar2 <- round(tax.pcoa$eig[2] / sum(tax.pcoa$eig), 3) * 100
xplainvar3 <- round(tax.pcoa$eig[3] / sum(tax.pcoa$eig), 3) * 100
sum.eig2 <- sum(xplainvar1, xplainvar2, xplainvar3)


# Define Plot Parameters
par(mar = c(5, 5, 1, 2) + 0.1)

# Initiate Plot
plot(tax.pcoa$points[ ,1], tax.pcoa$points[ ,2],
     xlim = c(-0.2, 0.2), ylim = c(-.16, 0.16),
     xlab = paste("PCoA 1 (", xplainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", xplainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(tax.pcoa$points[ ,1], tax.pcoa$points[ ,2],
       pch = 19, cex = 3, bg = "gray", col = "gray")
text(tax.pcoa$points[ ,1], tax.pcoa$points[ ,2], 
     labels = row.names(tax.pcoa$points))



```


***Question 5***: Using a combination of visualization tools and percent variation explained, how does the phylogenetically based ordination compare or contrast with the taxonomic ordination? 
What does this tell you about the importance of phylogenetic information in this system?

> ***Answer 5***: More varibility was explained by the taxonomic ordination (PCoA 1 = 28.4%, PCoA 2 = 12%) than for the phylogenetic ordination (PCoA 1 = 9.5%, PCoA 2 = 6%). Also, the groupings of the sites vary between the taxonomic and phylogenetic ordination (e.g., BC 105 and BC 108 are clumped alone together for the taxonomic but not the phylogenetic ordination). This shows that just because in this system two sites are similar taxonomically, they may not be phylogenetically similar. 

### C. Hypothesis Testing

**i. Categorical Approach**

In the R code chunk below, do the following:  
1. test the hypothesis that watershed has an effect on the phylogenetic diversity of bacterial communities.

```{r}
# Define Environmental Category
watershed <- env$Location

# Run PERMANOVA with `adonis()` Function 
adonis(dist.uf ~ watershed, permutations = 999)

# We can compare to PERMANOVA results based on taxonomy
adonis(
  vegdist(                             # create a distance matrix on
    decostand(comm, method = "log"),   # log-transformed relative abundances
    method = "bray") ~ watershed,      # using Bray-Curtis dissimilarity metric
  permutations = 999)

```

**ii. Continuous Approach**

In the R code chunk below, do the following:
1. from the environmental data matrix, subset the variables related to physical and chemical properties of the ponds, and  
2. calculate environmental distance between ponds based on the Euclidean distance between sites in the environmental data matrix (after transforming and centering using `scale()`).

```{r}
# Define environmental variables related to physical and chemical properties
envs <- env[, 5:19]

# Remove redudnant variables
envs <- envs[, -which(names(envs) %in% c("TDS", "Salinity", "Cal_Volume"))]

# Create distance matrix for environmental variables
env.dist <- vegdist(scale(envs), method = "euclid")

```

In the R code chunk below, do the following:  
1. conduct a Mantel test to evaluate whether or not UniFrac distance is correlated with environmental variation.

```{r}
# Conduct Mantel Test 
mantel(dist.uf, env.dist) 


```

Last, conduct a distance-based Redundancy Analysis (dbRDA). 

In the R code chunk below, do the following:  
1. conduct a dbRDA to test the hypothesis that environmental variation effects the phylogenetic diversity of bacterial communities,  
2. use a permutation test to determine significance, and
3. plot the dbRDA results

```{r}
# Conduct dbRDA (`vegan`)
ponds.dbrda <- vegan::dbrda(dist.uf ~ ., data = as.data.frame(scale(envs)))
# Permutation tests: axes and environmental variables
anova(ponds.dbrda, by = "axis")
ponds.fit <- envfit(ponds.dbrda, envs, perm = 999)
ponds.fit
# Calculate explained variation
dbrda.explainvar1 <- round(ponds.dbrda$CCA$eig[1] / 
                         sum(c(ponds.dbrda$CCA$eig, ponds.dbrda$CA$eig)), 3) * 100
dbrda.explainvar2 <- round(ponds.dbrda$CCA$eig[2] / 
                         sum(c(ponds.dbrda$CCA$eig, ponds.dbrda$CA$eig)), 3) * 100
# Make dbRDA plot
# Define plot parameters
par(mar = c(5, 5, 4, 4) + 0.1)

# Initiate plot
plot(scores(ponds.dbrda, display = "wa"), xlim = c(-2, 2), ylim = c(-2, 2),
     xlab = paste("dbRDA 1 (", dbrda.explainvar1, "%)", sep = ""),
     ylab = paste("dbRDA 2 (", dbrda.explainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, axes = FALSE)

# Add axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add points & labels
points(scores(ponds.dbrda, display = "wa"),
       pch = 19, cex = 3, bg = "gray", col = "gray")
text(scores(ponds.dbrda, display = "wa"), 
    labels = row.names(scores(ponds.dbrda, display = "wa")), cex = 0.5)

# Add environmental vectors
vectors <- scores(ponds.dbrda, display = "bp")

#row.names(vectors) <- c("Temp", "DO", "chla", "DON")
arrows(0, 0, vectors[,1] * 2, vectors[, 2] * 2, 
       lwd = 2, lty = 1, length = 0.2, col = "red")
text(vectors[,1] * 2, vectors[, 2] * 2, pos = 3, 
     labels = row.names(vectors))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.2, las = 1, col = "red", lwd = 2.2,
     at = pretty(range(vectors[, 1])) * 2, labels = pretty(range(vectors[, 1])))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.2, las = 1, col = "red", lwd = 2.2,
     at = pretty(range(vectors[, 2])) * 2, labels = pretty(range(vectors[, 2])))

```

***Question 6***: Based on the multivariate procedures conducted above, describe the phylogenetic patterns of $\beta$-diversity for bacterial communities in the Indiana ponds.   

> ***Answer 6***: Based on the permanova, it appears that there is a significant difference between the watersheds with respect to phylogenetic patterns. The Mantel's test suggests that there is tentative evidence (p = .054) to reject the null hypothesis and conclude that UniFrac distance is related to environmental conditions. However, not much variation is explained by the dbRDA, and none of the analyzed environmental conditions appear to have an especially big impact on the communities.

## 6) SPATIAL PHYLOGENETIC COMMUNITY ECOLOGY

### A. Phylogenetic Distance-Decay (PDD)

A distance decay (DD) relationship reflects the spatial autocorrelation of community similarity.
That is, communities located near one another should be more similar to one another in taxonomic composition than distant communities.
(This is analagous to the isolation by distance (IBD) pattern that is commonly found when examining genetic similarity of a populations as a function of space.)
Historically, the two most common explanations for the taxonomic DD are that it reflects spatially autocorrelated environmental variables and the influence of dispersal limitation. 
However, if phylogenetic diversity is also spatially autocorrelated, then evolutionary history may also explain some of the taxonomic DD pattern.
Here, we will construct the phylogenetic distance-decay (PDD) relationship

First, calculate distances for geographic data, taxonomic data, and phylogenetic data among all unique pair-wise combinations of ponds.

In the R code chunk below, do the following:  
1. calculate the geographic distances among ponds,  
2. calculate the taxonomic similarity among ponds,  
3. calculate the phylogenetic similarity among ponds, and  
4. create a dataframe that includes all of the above information. 

```{r}
# Geographic distances (km) among ponds
long.lat <- as.matrix(cbind(env$long, env$lat))
coord.dist <- earth.dist(long.lat, dist = TRUE)

# Taxonomic similarity among ponds (Bray-Curits distance)
bray.curtis.dist <- 1 - vegdist(comm) 

# Phylogenetic similarity among ponds (UniFrac)
unifrac.dist <- 1 - dist.uf 
  
# Transform all distances into list format:
unifrac.dist.ls <- liste(unifrac.dist, entry = "unifrac")
bray.curtis.dist.ls <- liste(bray.curtis.dist, entry = "bray.curtis")
coord.dist.ls <- liste(coord.dist, entry = "geo.dist")
env.dist.ls <- liste(env.dist, entry = "env.dist")


#Create data frame
df <- data.frame(coord.dist.ls, bray.curtis.dist.ls[, 3], unifrac.dist.ls[, 3],
                 env.dist.ls[, 3])
names(df)[4:6] <- c("bray.curtis", "unifrac", "env.dist")





```

Now, let's plot the DD relationships:  
In the R code chunk below, do the following:  
1. plot the taxonomic distance decay relationship,   
2. plot the phylogenetic distance decay relationship, and  
3. add trend lines to each.

```{r}
# Set initial plot parameters
par(mfrow=c(2, 1), mar = c(1, 5, 2, 1) + 0.1, oma = c(2, 0, 0, 0))

# Make plot for taxonomic DD
plot(df$geo.dist, df$bray.curtis, xlab = "", xaxt = "n", las = 1, ylim = c(0.1, 0.9),
     ylab="Bray-Curtis Similarity", 
     main = "Distance Decay", col = "SteelBlue")

# Regression for taxonomic DD 
DD.reg.bc <- lm(df$bray.curtis ~ df$geo.dist)
summary(DD.reg.bc)
abline(DD.reg.bc , col = "red4", lwd = 2)

# New plot parameters
par(mar = c(2, 5, 1, 1) + 0.1)

# Make plot for phylogenetic DD
plot(df$geo.dist, df$unifrac, xlab = "", las = 1, ylim = c(0.1, 0.9),
     ylab = "Unifrac Similarity", col = "darkorchid4")

# Regression for phylogenetic DD
DD.reg.uni <- lm(df$unifrac ~ df$geo.dist)
summary(DD.reg.uni)
abline(DD.reg.uni, col = "red4", lwd = 2)

# Add x-axis label to plot
mtext("Geographic Distance (km)", side = 1, adj = 0.55, 
      line = 0.5, outer = TRUE)




```

In the R code chunk below, test if the trend lines in the above distance decay relationships are different from one another.

```{r}
diffslope(df$geo.dist, df$unifrac, df$geo.dist, df$bray.curtis)

```

***Question 7***: Interpret the slopes from the taxonomic and phylogenetic DD relationships. If there are differences, hypothesize why this might be.   

> ***Answer 7***: There is a significant distance between the two slopes. While the distance decay for Bray-Curtis is noticeable on the plot, the distance decay for Unifrac is imperceptible. I hypothesize that evolutionary history has not significantly affected differences in distribution of OTUs at the sites, but rather that a changing environment has extirpated from some sites species that at one time had been at all of the sites.


## SYNTHESIS

Ignoring technical or methodological constraints, discuss how phylogenetic information could be useful in your own research. 
Specifically, what kinds of phylogenetic data would you need? How could you use it to answer important questions in your field? 
In your response, feel free to consider not only phylogenetic approaches related to phylogenetic community ecology, but also those we discussed last week in the PhyloTraits module, or any other concepts that we have not covered in this course.

> I'm most interested in conservation biology, so incorporating the idea of phylogenetic diversity in investigations of biodiversity could be very helpful. I now understand that, when thinking about how ecosystems have been and will be impacted by human disurbance, it is important to consider phylogenetic diversity given that a more phyolgenetically diverse community likely will be more resilient. I have written reports for entities that contracted me to describe avifaunal communities in the past without reporting on phylogenetic diversity (just taxonomic diversity), but in the future I'll know the importance of considering phylogenetic diversity when reporting. In order to do this, I'll only need access to GenBank so that I can acquuire sequences to construct a phylogenetic tree of the species that I observed at a site, and the knowledge that I've gained from this course.This knowledge could ultimately help, say, a conservation organization decide which property to acquire. 